src/modules/river/layout.cpp

```diff
@@ -0,0 +1,174 @@
+                                const char *title) {
+  // Intentionally empty
+}
+
+static void listen_mode(void *data, struct zriver_seat_status_v1 *zriver_seat_status_v1,
+                        const char *mode) {
+  // Intentionally empty
+}
+
+static const zriver_output_status_v1_listener output_status_listener_impl{
+    .focused_tags = listen_focused_tags,
+    .view_tags = listen_view_tags,
+    .urgent_tags = listen_urgent_tags,
+    .layout_name = listen_layout_name,
+    .layout_name_clear = listen_layout_name_clear,
+};
+
+static const zriver_seat_status_v1_listener seat_status_listener_impl{
+    .focused_output = listen_focused_output,
+    .unfocused_output = listen_unfocused_output,
+    .focused_view = listen_focused_view,
+    .mode = listen_mode,
+};
+
+static void handle_global(void *data, struct wl_registry *registry, uint32_t name,
+                          const char *interface, uint32_t version) {
+  if (std::strcmp(interface, zriver_status_manager_v1_interface.name) == 0) {
+    version = std::min<uint32_t>(version, 4);
+
+    // implies ZRIVER_OUTPUT_STATUS_V1_LAYOUT_NAME_CLEAR_SINCE_VERSION
+    if (version < ZRIVER_OUTPUT_STATUS_V1_LAYOUT_NAME_SINCE_VERSION) {
+      spdlog::error(
+          "river server does not support the \"layout_name\" and \"layout_clear\" events; the "
+          "module will be disabled" +
+          std::to_string(version));
+      return;
+    }
+    static_cast<Layout *>(data)->status_manager_ = static_cast<struct zriver_status_manager_v1 *>(
+        wl_registry_bind(registry, name, &zriver_status_manager_v1_interface, version));
+  }
+
+  if (std::strcmp(interface, wl_seat_interface.name) == 0) {
+    version = std::min<uint32_t>(version, 1);
+    static_cast<Layout *>(data)->seat_ = static_cast<struct wl_seat *>(
+        wl_registry_bind(registry, name, &wl_seat_interface, version));
+  }
+}
+
+static void handle_global_remove(void *data, struct wl_registry *registry, uint32_t name) {
+  // Nobody cares
+}
+
+static const wl_registry_listener registry_listener_impl = {.global = handle_global,
+                                                            .global_remove = handle_global_remove};
+
+Layout::Layout(const std::string &id, const waybar::Bar &bar, const Json::Value &config)
+    : waybar::ALabel(config, "layout", id, "{}"),
+      status_manager_{nullptr},
+      seat_{nullptr},
+      bar_(bar),
+      output_status_{nullptr} {
+  struct wl_display *display = Client::inst()->wl_display;
+  struct wl_registry *registry = wl_display_get_registry(display);
+  wl_registry_add_listener(registry, &registry_listener_impl, this);
+  wl_display_roundtrip(display);
+
+  output_ = gdk_wayland_monitor_get_wl_output(bar_.output->monitor->gobj());
+
+  if (!status_manager_) {
+    spdlog::error("river_status_manager_v1 not advertised");
+    return;
+  }
+
+  if (!seat_) {
+    spdlog::error("wl_seat not advertised");
+  }
+
+  label_.hide();
+  ALabel::update();
```

Comment:
This does not initially hide the label. I did notice the same behaviour from window... any ideas?

Reply:
Maybe because the label isn't empty, and so ALabel::update re-show it 🤔